packages/arb-token-bridge-ui/src/token-bridge-sdk/hooks/useArbTokenBridge.ts

```diff
@@ -54,6 +67,20 @@ export function getExecutedMessagesCacheKey({
     : `l2ChainId: ${l2ChainId}, position: ${event.position.toString()}`
 }
 
+export function getUniqueIdOrHashFromEvent(
+  event: L2ToL1EventResult
+): BigNumber {
+  const nitroEvent = event as EventArgs<NitroL2ToL1TransactionEvent>
+  const classicEvent = event as EventArgs<ClassicL2ToL1TransactionEvent>
+
+  // Nitro
+  if (typeof nitroEvent.hash !== 'undefined') {
```

Comment:
Not a fan of the casting here. If we cast to the type, `hash` is present in the type so it can't be `undefined`, giving as a false sense of security. I would revert to the previous implementation.

Reply:
Reverted